pkg/thanos/statefulset_test.go

```diff
@@ -327,6 +327,86 @@ func TestObjectStorageFile(t *testing.T) {
 	}
 }
 
+func TestAlertRelabel(t *testing.T) {
+	testKey := "thanos-alertrelabel-config-secret"
+
+	sset, err := makeStatefulSet(&monitoringv1.ThanosRuler{
+		ObjectMeta: metav1.ObjectMeta{},
+		Spec: monitoringv1.ThanosRulerSpec{
+			QueryEndpoints: emptyQueryEndpoints,
+			AlertRelabelConfigs: &v1.SecretKeySelector{
+				Key: testKey,
+			},
+		},
+	}, defaultTestConfig, nil, "")
+	if err != nil {
+		t.Fatalf("Unexpected error while making StatefulSet: %v", err)
+	}
+
+	if sset.Spec.Template.Spec.Containers[0].Name != "thanos-ruler" {
+		t.Fatalf("expected 1st containers to be thanos-ruler, got %s", sset.Spec.Template.Spec.Containers[0].Name)
+	}
+
+	var containsEnvVar bool
+	for _, env := range sset.Spec.Template.Spec.Containers[0].Env {
+		if env.Name == "AM_RELABEL_CONFIG" {
+			if env.ValueFrom.SecretKeyRef.Key == testKey {
+				containsEnvVar = true
+				break
+			}
+		}
+	}
+	if !containsEnvVar {
+		t.Fatalf("Thanos ruler is missing expected AM_RELABEL_CONFIG env var with correct value")
+	}
+
+	{
+		var containsArg bool
+		const expectedArg = "--alert.relabel-config=$(AM_RELABEL_CONFIG)"
+		for _, arg := range sset.Spec.Template.Spec.Containers[0].Args {
+			if arg == expectedArg {
+				containsArg = true
+				break
+			}
+		}
+		if !containsArg {
+			t.Fatalf("Thanos ruler is missing expected argument: %s", expectedArg)
+		}
+	}
+}
+
+func TestAlertRelabelFile(t *testing.T) {
+	testPath := "/vault/secret/config.yaml"
+	sset, err := makeStatefulSet(&monitoringv1.ThanosRuler{
+		ObjectMeta: metav1.ObjectMeta{},
+		Spec: monitoringv1.ThanosRulerSpec{
+			QueryEndpoints:         emptyQueryEndpoints,
+			AlertRelabelConfigFile: &testPath,
```

Comment:
could you add a `AlertRelabelConfigs` field to verify that `AlertRelabelConfigFile` takes precedence?